fedimint-server/src/consensus/mod.rs

```diff
@@ -107,105 +58,85 @@ impl FedimintConsensus {
-            dbtx.set_tx_savepoint()
-                .await
-                .expect("Setting transaction savepoint failed");
+        let new_session_index = dbtx.get_value(&SessionIndexKey).await.unwrap_or(0) + 1;
 
-            if let ConsensusItem::Transaction(ref transaction) = consensus_item {
-                if !processed_txes.insert(transaction.tx_hash()) {
-                    continue;
-                }
-            }
+        dbtx.insert_entry(&SessionIndexKey, &new_session_index)
+            .await;
 
-            match self
-                .process_consensus_item(&mut dbtx, consensus_item.clone(), peer_id)
-                .await
-            {
-                Ok(()) => {
-                    debug!(
-                        target: LOG_CONSENSUS,
-                        "Accept consensus item from {peer_id}"
-                    );
-                }
-                Err(error) => {
-                    debug!(
-                        target: LOG_CONSENSUS,
-                        "Discard consensus item from {peer_id}: {error}"
-                    );
-
-                    dbtx.rollback_tx_to_savepoint()
-                        .await
-                        .expect("Rolling back transaction to savepoint failed");
-
-                    if let ConsensusItem::Transaction(transaction) = consensus_item {
-                        rejected_txs.insert(transaction.tx_hash());
-                    }
-                }
-            }
-        }
+        dbtx.remove_by_prefix(&AcceptedIndexPrefix).await;
 
-        if let Some(reference_rejected_txs) = reference_rejected_txs.as_ref() {
-            // Result of the consensus are supposed to be deterministic.
-            // If our result is not the same as what the (honest) majority of the federation
-            // signed over, it's a catastrophic bug/mismatch of Federation's fedimintd
-            // implementations.
-            assert_eq!(
-                reference_rejected_txs, &rejected_txs,
-                "rejected_txs mismatch: reference = {reference_rejected_txs:?} != {rejected_txs:?}"
-            );
-        }
+        dbtx.commit_tx_result()
+            .await
+            .expect("This is the only place we write to those keys");
+    }
 
-        let epoch_history = self
-            .save_epoch_history(consensus_outcome, &mut dbtx, &mut vec![], rejected_txs)
-            .await;
+    pub async fn process_consensus_item(
+        &self,
+        item: Vec<u8>,
+        item_index: u64,
+        peer_id: PeerId,
+    ) -> anyhow::Result<()> {
+        let _timing /* logs on drop */ = timing::TimeReporter::new("process_consensus_item");
 
-        dbtx.commit_tx_result()
+        let mut reader = std::io::Cursor::new(item);
+        let consensus_item = ConsensusItem::consensus_decode(&mut reader, &self.decoders())?;
+
+        let item_debug = debug::item_message(&consensus_item);
+        debug!("\n  Peer {peer_id}: {item_debug}");
```

Comment:
`\n`?

Reply:
copy and pasted that from the previous code